src/ansible_navigator/ui_framework/utils.py

```diff
@@ -3,28 +3,48 @@
 import functools
 import re
 
+from dataclasses import is_dataclass
 from math import floor
+from typing import Any
+from typing import Dict
 from typing import List
+from typing import Union
 
 
-def convert_percentage(data_dict: dict, keys: List, progress_bar_width: int) -> None:
-    """convert a string % to a little progress bar
-    not recursive
-    80% = 80%|XXXXXXXX  |
+def convert_percentage(
+    content: Union[Dict[str, Any], Any],
```

Comment:
This can simplify to `Any` for now, since `A ∪ ⊤` ~ `T`

(where T is the [top type](https://en.wikipedia.org/wiki/Top_type))

Reply:
The Any there is just a place holder for ContentBase in #1014